src/normal_form_game.jl

```diff
@@ -612,3 +612,50 @@ function pure2mixed(num_actions::Integer, action::PureAction)
     mixed_action[action] = 1
     return mixed_action
 end
+
+# is_pareto_efficient & is_pareto_dominant
+
+for args = ((:is_pareto_efficient, &), (:is_pareto_dominant, |))
```

Comment:
I think `&&` and `||` should be used rather than the bitwise operators `&` and `|`, while I see `&&` and `||` do not work with `@eval`. What about defining functions:
```jl
_and(a, b) = a && b
_or(a, b) = a || b
```

Reply:
This is not a good idea...
Your original idea seems to be the way to go. What about:

```jl
function pareto_inferior_to(payoff_profile1, payoff_profile2)
    all(payoff_profile2 .>= payoff_profile1) &&
    any(payoff_profile2 .> payoff_profile1)
end

function not_pareto_superior_to(payoff_profile1, payoff_profile2)
    any(payoff_profile2 .> payoff_profile1) ||
    all(payoff_profile2 .== payoff_profile1)
end
```

Reply:
Just to be clear, I should get rid of `is_pareto_efficient` and `is_pareto_dominant` and replace them by `pareto_inferior_to` and `not_pareto_superior_to` (including the tests)?

Reply:
No. I was referring to the operations in the `if` clause.